### ISSUE TITLE
Lit change-in-update warnings from many components

### ISSUE THREAD

**Issue by medmunds** [shoelace-style/webawesome#1269]
### Describe the bug
When running in dev mode, many wa-* components cause Lit to warn about "scheduled an update after an update completed" when first rendered.

The equivalent Shoelace components did not issue this warning.

### To Reproduce

1. Create a custom LitElement that renders some Web Awesome components, e.g., wa-button
2. Run a development server and view your project
3. Examine the browser console:

Results:
> Element wa-button scheduled an update (generally because a property was set) after an update completed, causing a new update to be scheduled. This is inefficient and should be avoided unless the next update can only be scheduled as a side effect of the previous update. See https://lit.dev/msg/change-in-update for more information.

Expected: did not generally occur with Shoelace. (There were some specific component/property value combos that could cause that warning, but not just rendering components in the initial state.)

### Demo

Here's a codepen that loads Lit in dev mode and renders wa-button and sl-button side by side. Notice the console output warns about wa-button but not sl-button.
https://codepen.io/mvedmunds/pen/azvWxYq?editors=0011

### Screenshots

<img width="779" height="519" alt="Image" src="https://github.com/user-attachments/assets/8d347ca2-dfb4-48d7-8ee5-a24d6f3366dd" />

### More information
At a minimum, these components are affected in 3.0.0.beta-4:
- wa-button
- wa-card
- wa-checkbox
- wa-input
- wa-radio
- wa-radio-group
- wa-select
- wa-slider

(I'm using lit@3.3.1, but saw the same warning with earlier versions.)

**Comment by claviska** [comment]
@KonnorRogers is dev mode not enabled in our development environment? 🤔

**Comment by KonnorRogers** [comment]
@claviska Nope. and its been this way since Shoelace. I can add it.

```js
const config = {
  // ... other esbuild config
  conditions: ["development"]
}
```

https://esbuild.github.io/api/#conditions

https://lit.dev/docs/tools/development/#development-and-production-builds

**Comment by claviska** [comment]
Thanks, @KonnorRogers!

@medmunds We can look into optimizing the offending components, but I believe there were a handful that were legit and won't be suppressed.

I have no idea why dev mode has been disabled for so long 🙈

**Comment by KonnorRogers** [comment]
@claviska I think we turned it off because the output from Lit looks like this:

https://github.com/user-attachments/assets/f4d51ffc-46f6-4cdd-99b4-06cb6626a1de

**Comment by KonnorRogers** [comment]
I actually started a quick branch here:

https://github.com/shoelace-style/webawesome/pull/1284

that resolved about 99% of the update warnings.

(in the test suite theres a couple warnings, but I think they're related to test suite things triggering it rather than actual component code, would have to click around more to check)


i just need to figure out how to different the "exportConditions" in ESBuild so we ship production builds for CDN users and for our production site.

**Comment by medmunds** [comment]
> I believe there were a handful that were legit and won't be suppressed

Lit has a documented way to [suppress this warning for an individual component](https://lit.dev/docs/tools/development/#development-build-runtime-warnings) that legitimately needs update to retrigger an update:

```typescript
WaComponentThatLegitReUpdates.disableWarning?.("change-in-update");
```

There's also an undocumented ([but tested](https://github.com/lit/lit/blob/1889e0eb1cdda28a9e9981ac4657d381e1d1bfbf/packages/lit-html/src/test/lit-html_dev_mode_disable_test.ts#L15-L16)) way to suppress the "Lit is in dev mode" warning if it's muddying up your test results:

```typescript
if (however.you.detect.TEST_MODE) {
  globalThis.litIssuedWarnings ??= new Set();
  globalThis.litIssuedWarnings!.add('dev-mode');
}
```

**Comment by claviska** [comment]
TIL; thanks for that, @medmunds!

**Comment by claviska** [comment]
Fixed in #2161
